app/src/main/java/de/westnordost/streetcomplete/screens/settings/SettingsFragment.kt

```diff
@@ -144,10 +138,26 @@ class SettingsFragment :
         }
     }
 
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setUpToolbarTitleAndIcon(view.findViewById(R.id.toolbar))
+    }
+
     override fun onStart() {
         super.onStart()
-        findPreference<Preference>("quests")?.summary = getQuestPreferenceSummary()
-        findPreference<Preference>("quest_presets")?.summary = getQuestPresetsPreferenceSummary()
+
+        setQuestPreferenceSummary()
+        setQuestPresetsPreferenceSummary()
+
+        visibleQuestTypeSource.addListener(this)
+        questPresetsSource.addListener(this)
```

Comment:
Here and probably in the other fragments:

The pattern I try to follow more or less consistently in this project is to have anonymous objects for listeners like this instead of making the whole class a listener. There is not really a strong reason for this other than consistency, the (weak) reasons are:
- the data sources the fragment is listening to is part of the internals of the class, but this makes this be exposed on the public interface
- In Java/Kotlin, I find it clearer to organize code in this way. I.e. if there are just some `override fun onSomethingSomething` functions floating around, one does not immediately know where those are coming from. An anonymous object as a field is also something that can be just collapsed in the IDE or editor of choice

A (weak) reason against this pattern would be that it is more complicated to write unit tests involving listening to these data sources, but it doesn't apply here because the UI code is not unit tested in general in this app.